core/src/main/scala/org/apache/spark/rdd/ReliableCheckpointRDD.scala

```diff
@@ -239,7 +239,14 @@ private[spark] object ReliableCheckpointRDD extends Logging {
       val fs = partitionerFilePath.getFileSystem(sc.hadoopConfiguration)
       val fileInputStream = fs.open(partitionerFilePath, bufferSize)
       val serializer = SparkEnv.get.serializer.newInstance()
-      val deserializeStream = serializer.deserializeStream(fileInputStream)
+      val deserializeStream = try {
+        serializer.deserializeStream(fileInputStream)
+      } catch {
+        case e : Throwable =>
+          fileInputStream.close()
```

Comment:
Yes I think that's a good idea. Do you need the generic types on the "try" methods, even?